portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +35,42 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello stranger! Welcome to my portfolio");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    for (Entity entity : results.asIterable()) {
+      String text = (String) entity.getProperty("text");
+      String author = (String) entity.getProperty("author");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(text, author, timestamp);
+      comments.add(comment);
+    }
+
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentString = request.getParameter("comment-input");
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", commentString);
+    commentEntity.setProperty("author", "anonymous");
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
There should be comments somewhere so I understand that Datastore is being used to store user messages without finding it in your code. You should consider either function-level comments or class-level comments (see https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required)

Reply:
Done!

Reply:
Thanks for the extra comments! I'd still like you to add either class-level or function-level comments to clarify what is happening in this servlet, so if someone were to look at this servlet they would know what it does without having to look into the code.

Reply:
Alright, I made sure to have comments on every class and method.

Reply:
Perfect, thanks Shaar! I know it might seem tedious, but if someone else is looking at your code, they might go straight to the method and without the comment, they won't know what's going on. Or they'll look for that class-level documentation. They should be able to look in any of these places and get clarification as to what the code in that file is doing.

Thanks again for the changes!